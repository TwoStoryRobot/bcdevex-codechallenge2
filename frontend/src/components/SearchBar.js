import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import Paper from '@material-ui/core/Paper'
import FormControl from '@material-ui/core/FormControl'
import Input from '@material-ui/core/Input'
import InputAdornment from '@material-ui/core/InputAdornment'
import IconButton from '@material-ui/core/IconButton'
import SearchIcon from 'mdi-react/SearchIcon'

const PaddedPaper = styled(Paper)`
  padding: 0 32px;
`

class SearchBar extends React.Component {
  render() {
    const searchAdornment = (
      <InputAdornment position="end">
        <IconButton>
          <SearchIcon />
        </IconButton>
      </InputAdornment>
    )
    return (
      <PaddedPaper>
        <FormControl>
          <Input
            disableUnderline
            placeholder="Filter"
            endAdornment={searchAdornment}
          />
        </FormControl>
      </PaddedPaper>
    )
  }
}

export default SearchBar
