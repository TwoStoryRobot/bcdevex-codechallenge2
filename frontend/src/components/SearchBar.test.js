import React from 'react'
import { render, cleanup, fireEvent } from 'react-testing-library'
import 'jest-dom/extend-expect'

import SearchBar from './SearchBar'

afterEach(cleanup)

describe('SearchBar', () => {
  it('should render with required props', async () => {
    const { container } = render(<SearchBar />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
