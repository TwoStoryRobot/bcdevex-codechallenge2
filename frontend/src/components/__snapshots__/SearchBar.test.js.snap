// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render with required props 1`] = `
<div
  class="MuiPaper-root-01 MuiPaper-elevation2-05 MuiPaper-rounded-02 sc-bdVaJa bVdUDK"
>
  <div
    class="MuiFormControl-root-028"
  >
    <div
      class="MuiInput-root-032 MuiInput-formControl-033"
    >
      <input
        aria-invalid="false"
        class="MuiInput-input-040"
        placeholder="Filter"
        type="text"
        value=""
      />
      <div
        class="MuiInputAdornment-root-045 MuiInputAdornment-positionEnd-047"
      >
        <button
          tabindex="0"
          class="MuiButtonBase-root-054 MuiIconButton-root-048"
          type="button"
        >
          <span
            class="MuiIconButton-label-053"
          >
            <svg
              class="mdi-icon "
              width="24"
              height="24"
              fill="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M9.5,3C13.09,3 16,5.91 16,9.5C16,11.11 15.41,12.59 14.44,13.73L14.71,14H15.5L20.5,19L19,20.5L14,15.5V14.71L13.73,14.44C12.59,15.41 11.11,16 9.5,16C5.91,16 3,13.09 3,9.5C3,5.91 5.91,3 9.5,3M9.5,5C7,5 5,7 5,9.5C5,12 7,14 9.5,14C12,14 14,12 14,9.5C14,7 12,5 9.5,5Z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root-057"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
